File: mobile-app/lib/models/healthcare_facility.dart
class HealthcareFacility {
  final int id;
  final String name;
  final String type;
  final String? services;
  final String? address;
  final double? latitude;
  final double? longitude;
  final String? contactNumber;
  final String? operatingHours;
  final bool? freeServices;
  final bool isActive;

  HealthcareFacility({
    required this.id,
    required this.name,
    required this.type,
    this.services,
    this.address,
    this.latitude,
    this.longitude,
    this.contactNumber,
    this.operatingHours,
    this.freeServices,
    this.isActive = true,
  });

  factory HealthcareFacility.fromJson(Map<String, dynamic> json) {
    return HealthcareFacility(
      id: json['id'] as int,
      name: json['name'] as String,
      type: json['type'] as String,
      services: json['services'] as String?,
      address: json['address'] as String?,
      latitude: json['latitude'] as double?,
      longitude: json['longitude'] as double?,
      contactNumber: json['contactNumber'] as String?,
      operatingHours: json['operatingHours'] as String?,
      freeServices: json['freeServices'] as bool?,
      isActive: json['isActive'] as bool? ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'type': type,
      'services': services,
      'address': address,
      'latitude': latitude,
      'longitude': longitude,
      'contactNumber': contactNumber,
      'operatingHours': operatingHours,
      'freeServices': freeServices,
      'isActive': isActive,
    };
  }
}

File: backend/src/main/java/com/civicbridge/model/HealthcareFacility.java
package com.civicbridge.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "healthcare_facilities")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class HealthcareFacility {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    private String type; // HOSPITAL, CLINIC, VACCINATION_CENTER
    
    @Column(length = 1000)
    private String services;
    
    private String address;
    
    private Double latitude;
    
    private Double longitude;
    
    private String contactNumber;
    
    private String operatingHours;
    
    private Boolean freeServices;
    
    @Column(name = "is_active")
    private Boolean isActive = true;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
